academy/xl-cosmjs/multi-msg.md

```diff
@@ -0,0 +1,508 @@
+
+## Token transfer messages
+
+In order to use `signAndBroadcast` to send tokens, you need to figure out what messages go into the `messages: readonly EncodeObject[]`. Examine the `sendTokens` function body:
+
+```typescript [https://github.com/cosmos/cosmjs/blob/7aad551/packages/stargate/src/signingstargateclient.ts#L224-L232]
+const sendMsg: MsgSendEncodeObject = {
+    typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+    value: {
+      fromAddress: senderAddress,
+      toAddress: recipientAddress,
+      amount: [...amount],
+    },
+  };
+return this.signAndBroadcast(senderAddress, [sendMsg], fee, memo);
+```
+
+Therefore, when sending back to the faucet, instead of calling:
+
+```typescript
+const result = await signingClient.sendTokens(
+    alice,
+    faucet,
+    [{ denom: "uatom", amount: "100000" }],
+    {
+        amount: [{ denom: "uatom", amount: "500" }],
+        gas: "200000",
+    },
+)
+```
+
+Alice can instead call:
+
+```typescript
+const result = await signingClient.signAndBroadcast(
+    alice,
+    [
+        {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: alice,
+                toAddress: faucet,
+                amount: [
+                    { denom: "uatom", amount: "100000" },
+                ],
+            },
+          },
+    ],
+    {
+        amount: [{ denom: "uatom", amount: "500" }],
+        gas: "200000",
+    },
+)
+```
+
+You can confirm this by making the change in your `experiment.ts` from the previous section, and running it again.
+
+<HighlightBox type="tip">
+
+In fact, building a transaction in this way is recommended. `SigningStargateClient` offers you convenience methods such as `sendTokens` for simple use cases only.
+
+</HighlightBox>
+
+## What is this long string?
+
+Note the `typeUrl: "/cosmos.bank.v1beta1.MsgSend"` string. This comes from Protobuf and is a concatenation of:
+
+1. The `package` where `MsgSend` is initially declared:
+
+    ```protobuf [https://github.com/cosmos/cosmos-sdk/blob/3a1027c/proto/cosmos/bank/v1beta1/tx.proto#L2]
+    package cosmos.bank.v1beta1;
+    ```
+
+2. And the name of the message itself, `MsgSend`:
+
+    ```protobuf [https://github.com/cosmos/cosmos-sdk/blob/3a1027c/proto/cosmos/bank/v1beta1/tx.proto#L22]
+    message MsgSend {
+        ...
+    }
+    ```
+
+This is the canonical identifier of the type of the message serialized next to it. Additionally, it is made to be easy for you, the developer, to understand what it represents. The blockchain client knows how to serialize or deserialize it only because this `"/cosmos.bank.v1beta1.MsgSend"` string is passed along. With this `typeUrl`, the blockchain client and CosmJS are able to pick the right deserializer. This object is also named `MsgSend` in `cosmjs-types`.
```

Comment:
Seeing as this is word for word the same as the previous one, we might as well skip this. I didn't connect the dots myself earlier. Sorry about that!

```suggestion
```